core/include/tangram/map.h

```diff
@@ -230,8 +230,9 @@ class Map {
     // Get the tilt angle of the view in radians; 0 corresponds to straight down
     float getTilt();
 
-    // Get the CameraPosition that encloses the bounds given by _a and _b
-    CameraPosition getEnclosingCameraPosition(LngLat a, LngLat b, int buffer);
+    // Get the CameraPosition that encloses the bounds given by _a and _b and
+    // leaves at least the given amount of padding on each side (in logical pixels).
+    CameraPosition getEnclosingCameraPosition(LngLat _a, LngLat _b, int _leftPad, int _topPad, int _rightPad, int _bottomPad);
```

Comment:
Will it be better (cleaner) to pass in std::array here instead of explicit 4 padding dimensions?

Reply:
I'm pretty sure I'd mess up the order of these arguments if they weren't named. The better solution for consistency is probably to define a padding struct with named members.

Reply:
Yeah, maybe its a small fish in big ocean, but we do lose some cache coherence by breaking the well formed memory struct for these paddings.